src/loan/debt_order.ts

```diff
@@ -0,0 +1,463 @@
+import { BigNumber } from "../../utils/bignumber";
+
+import {
+    BLOCK_TIME_ESTIMATE_SECONDS,
+    NULL_ADDRESS,
+    NULL_ECDSA_SIGNATURE,
+    SALT_DECIMALS,
+} from "../../utils/constants";
+
+import { CollateralizedSimpleInterestLoanOrder } from "../adapters/collateralized_simple_interest_loan_adapter";
+
+import { Dharma } from "../types/dharma";
+
+import { DebtOrderDataWrapper } from "../wrappers";
+
+import {
+    DebtOrderData,
+    DurationUnit,
+    ECDSASignature,
+    EthereumAddress,
+    InterestRate,
+    TimeInterval,
+    TokenAmount,
+} from "../types";
+
+export const DEBT_ORDER_ERRORS = {
+    ALREADY_SIGNED_BY_DEBTOR: `The debtor has already signed this debt order.`,
+    ALREADY_SIGNED_BY_CREDITOR: `The creditor has already signed this debt order.`,
+};
+
+export interface LoanRequestConstructorParams {
```

Comment:
can we rename these interfaces to `DebtOrder...`?

tricky one will be LoanRequestData 😓